### PR TITLE
Add impact sample app. Fix some things while creating it

### PR DESCRIPTION
**Pull request by sergioboi**
The sample isn't ready. But I consider merging by small pieces

### CODE REVIEW

File: subprojects/gradle/impact-shared/src/main/java/com/avito/impact/ConfigurationType.kt [hunk 1]
```diff
@@ -0,0 +1,3 @@
+package com.avito.impact
+
+enum class ConfigurationType { IMPLEMENTATION, UNIT_TESTS, ANDROID_TESTS, LINT }
```
Comment (sergioboi) [inline subprojects/gradle/impact-shared/src/main/java/com/avito/impact/ConfigurationType.kt:3]: Consider it clear with new name

File: subprojects/gradle/impact-shared/src/main/java/com/avito/impact/configuration/AndroidTestConfiguration.kt [hunk 2]
```diff
@@ -9,14 +9,14 @@ class AndroidTestConfiguration(module: InternalModule) : SimpleConfiguration(mod
 
     override val isModified: Boolean by lazy {
         dependencies.any { it.isModified }
-            || module.testConfiguration.isModified
+            || module.implementationConfiguration.isModified
```
Comment (sergioboi) [inline subprojects/gradle/impact-shared/src/main/java/com/avito/impact/configuration/AndroidTestConfiguration.kt]: androidTest configuration extends implementation

File: subprojects/gradle/impact-shared/src/main/java/com/avito/impact/configuration/LintConfiguration.kt [hunk 3]
```diff
@@ -24,15 +24,11 @@ class LintConfiguration(module: InternalModule) : SimpleConfiguration(module) {
     }
 
     override val isModified: Boolean by lazy {
-        dependencies.any { it.isModified }
-            || module.androidTestConfiguration.isModified
-            || hasChangedFiles
+        dependencies.any { it.isModified } || hasChangedFiles
```
Comment (sergioboi) [inline subprojects/gradle/impact-shared/src/main/java/com/avito/impact/configuration/LintConfiguration.kt]: lint configuration extends nothing

File: subprojects/gradle/instrumentation-test-impact-analysis/src/main/kotlin/com/avito/instrumentation/impact/AnalyzeTestImpactTask.kt [hunk 4]
```diff
@@ -95,8 +95,11 @@ abstract class AnalyzeTestImpactTask @Inject constructor(
         val node = environmentInfo.node?.take(32) ?: "_"
         val prefix = graphiteSeries(envName, node)
 
-        val percentage = ((testsToRun.size.toFloat() / allTests.size) * 100).roundToInt()
+        // efficiency exists only if we have at least one test
+        if (allTests.isNotEmpty()) {
+            val percentage = ((testsToRun.size.toFloat() / allTests.size) * 100).roundToInt()
```
Comment (sergioboi) [inline subprojects/gradle/instrumentation-test-impact-analysis/src/main/kotlin/com/avito/instrumentation/impact/AnalyzeTestImpactTask.kt:100]: get an error if allTests == 0

File: subprojects/gradle/instrumentation-test-impact-analysis/src/main/kotlin/com/avito/instrumentation/impact/TestBytecodeAnalyzeAction.kt [hunk 5]
```diff
@@ -105,10 +105,13 @@ abstract class TestBytecodeAnalyzeAction : WorkAction<TestBytecodeAnalyzeAction.
 
     override fun execute() {
         val contextLoader = ContextLoader()
-        val foldersWithClassesToLoad = bytecodeResolver.resolveBytecodeWithoutDependencyToAnotherConfigurations(
-            reportType = ReportType.ANDROID_TESTS
+        val androidTestsBytecodeFolders = bytecodeResolver.resolveBytecodeWithoutDependencyToAnotherConfigurations(
+            configurationType = ConfigurationType.ANDROID_TESTS
         )
-        val context: Context = contextLoader.load(foldersWithClassesToLoad)
+        require(androidTestsBytecodeFolders.isNotEmpty()) {
```
Comment (sergioboi) [inline subprojects/gradle/instrumentation-test-impact-analysis/src/main/kotlin/com/avito/instrumentation/impact/TestBytecodeAnalyzeAction.kt:111]: fail fast with a clear message

**Review by dsvoronin — APPROVED** [review]

File: subprojects/gradle/impact-shared/src/main/java/com/avito/impact/configuration/AndroidTestConfiguration.kt (hunk 2 above)
Comment (sergioboi) [reply subprojects/gradle/impact-shared/src/main/java/com/avito/impact/configuration/AndroidTestConfiguration.kt]: Removed from that commit

File: subprojects/gradle/impact-shared/src/main/java/com/avito/impact/configuration/LintConfiguration.kt (hunk 3 above)
Comment (sergioboi) [reply subprojects/gradle/impact-shared/src/main/java/com/avito/impact/configuration/LintConfiguration.kt]: Removed from that commit

**Pull request merged**